apps/dashboard/app/apps/user_configuration.rb

```diff
@@ -0,0 +1,56 @@
+class UserConfiguration
+
+  def initialize
+    @config = ::Configuration.config
+  end
+
+  # The dashboard's landing page layout. Defaults to nil.
+  def dashboard_layout
+    fetch(:dashboard_layout, nil)
+  end
+
+  # The configured pinned apps
+  def pinned_apps
+    fetch(:pinned_apps, [])
+  end
+
+  # The length of the "Pinned Apps" navbar menu
+  def pinned_apps_menu_length
+    fetch(:pinned_apps_menu_length, 6)
+  end
+
+  # What to group pinned apps by
+  # @return [String, ""] Defaults to ""
+  def pinned_apps_group_by
+    group_by = fetch(:pinned_apps_group_by, "")
+
+    # FIXME: the user_configuration shouldn't really know the API of
+    # OodApp or subclasses. This is a hack because subclasses of OodApp overload
+    # the category and subcategory to something new while saving the original.
+    # The fix would be to move this knowledge to somewhere more appropriate than here.
+    if group_by == 'category' || group_by == 'subcategory'
+      "original_#{group_by}"
+    else
+      group_by
+    end
+  end
+
+  def profile_links
+    fetch(:profile_links, [])
+  end
+
+  def profile
+    CurrentUser.user_settings[:profile].to_sym if CurrentUser.user_settings[:profile]
+  end
+
+  private
+
+  def fetch(key_value, default_value = nil)
+    key = key_value ? key_value.to_sym : nil
+    profile_config = @config.dig(:profiles, profile) || {}
+
+    # Return the value if defined, regardless of type
+    # This is to mimic the Hash.fetch behaviour
+    profile_config.key?(key) ? profile_config[key] : @config.fetch(key, default_value)
+  end
+end
```

Comment:
Hi @johrstrom 
I found an issue when the profile overrides a value with `nil`. The UserConfiguration was not respecting the override and using the root or default values.

I discovered this when testing the `pinned_apps_group_by` functionality. Which I defined in the root config, but wanted to reset in a profile.

I added some new tests to check and the fix above. Let me know what you think

Reply:
Got it, you're trying to use `nil` as an value in the profile, so you're checking for the key. Maybe say something similar in the comment?

Reply:
I have updated the comment, but not sure it is clear what I mean.